lib/modules/versioning/unity3d/index.ts

```diff
@@ -0,0 +1,57 @@
+import { regEx } from '../../../util/regex';
+import { GenericVersion, GenericVersioningApi } from '../generic';
+import type { VersioningApi } from '../types';
+
+export const id = 'unity3d';
+export const displayName = 'Unity3D';
+export const urls = [
+  'https://docs.unity3d.com/Manual/ScriptCompilationAssemblyDefinitionFiles.html#version-define-expressions',
+];
+export const supportsRanges = false;
+
+class Unity3dVersioningApi extends GenericVersioningApi {
+  private static readonly parsingRegex = regEx(
+    /^(?<Major>\d+)\.(?<Minor>\d+)\.(?<Patch>\d+)(?<ReleaseStream>\w)(?<Build>\d+)/gm,
+  );
+
+  private static readonly ReleaseStreamType = new Map([
```

Comment:
do we really need this map? the keys are not really used.

Reply:
Only used the keys for readability, but I can see how this might be viewed as overengineered. Resolved it in 0b3f73dd00795f5babe63ebd1fbb4a0ce56bac26 and replaced it with comments instead.